### DISCUSSION TITLE
[Feature] Sort Options for timeline

### DISCUSSION THREAD

**Discussion by patrickap** [immich-app/immich#4170] [Feature Request]
### Sort Options

I’ve just installed the latest update v1.79.0 on my end and I must say, I’m quite impressed with the new library feature! A big thank you to everyone involved. I’m experiencing some issues with a few pictures, though. Since I don’t upload them using the app in this case, it’s possible that some of my (old) pictures do not have EXIF data. This causes the timeline order to be no longer correct. These pictures from my library appear at the top of the timeline and are a bit disruptive. I’ve been wondering if it wouldn’t make sense to make the timeline or the library sortable by different attributes. For instance, “Auto” - as it currently works, “Name” - based on file names, and perhaps more, with the option for ascending or descending order. All my pictures for example are named in a sortable format like „2010-07-19_11-49-02.jpg“ even though EXIF data is missing. This way, everyone could sort their timeline as they prefer. Since I’m not familiar with the implementation details, I’m curious about how complex this change would be. Are there any other opinions on my idea?

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by emhl** [comment]
Allowing to add a regular expression to determine the time from filename (if it's different enough from the file creation date) would be very helpful. photos that get shared via WhatsApp normally dont have any exif data, but the name easily shows when it was sent. and if you don't want to automatically import all images recieved vie Whatsapp you can easily get different dates

**Reply by patrickap** [reply]
That would be one idea. I tried to adjust the jpg metadata using exiftool. That would work, of course. However, with videos (mp4, mov), it becomes more complex. There is no unified standard for metadata in videos. Some of my videos are rendered completely wrong in the timeline. Then there are formats like png or gif that can also cause a lot of trouble. The latter doesn’t even have useful metadata I guess. Simply parsing the date from the filename would help a lot.

**Reply by patrickap** [reply]
In a certain photo app, I noticed that it allows you to define different rules for metadata extraction. You have the flexibility to arrange these rules in any order you prefer. For instance, one rule could be designed for extracting EXIF data, while another rule might focus on parsing the date from the file name, among other possibilities.

**Comment by patrickap** [comment]
> In a certain photo app, I noticed that it allows you to define different rules for metadata extraction. You have the flexibility to arrange these rules in any order you prefer. For instance, one rule could be designed for extracting EXIF data, while another rule might focus on parsing the date from the file name, among other possibilities.

What are your thoughts on this one, @alextran1502?

**Comment by Symbiot78** [comment]
I was looking for something on how immich handles files without exif and I am experiencing the same.. confusion.

Automatically sorting photos & videos without exif data ( or just without dateTaken) would be nice.

Another thing is that when there's no Exif data you cannot see the filename. Seeing the filename would make it a bit easier to backtrack to the original file and see if it's possible to figure out dates etc. 

just a thought.. I just started using immich so there's a lot I don't know yet.. but so far... kudos to the dev(s).

**Comment by aisbergde** [comment]
I use external software (digiKam) to correct date and time information, to extract from filename and so on and to write this into EXIF / sidecar. This works great and immich accepted the changes. I think it is better, to use external software to correct datetime, then having some workarounds in immich, based on filename.

**Comment by Symbiot78** [comment]
@aisbergde  -  thanks. However to get to the point where I can use digiKam I need a filename from immich :-)
So I can investigate IF I can find some more info.

**Comment by Symbiot78** [comment]
oh - and of course.. if I could single out all the photos without exif data easily in Immich that would help enormously.

**Comment by aisbergde** [comment]
> @aisbergde  -  thanks. However to get to the point where I can use digiKam I need a filename from immich :-)
> So I can investigate IF I can find some more info.
> 

I don't understand, what you mean. If you would use the current default naming template, then the original filename will be preserved. I don't know, what happened in the past. Maybe in the past another template was the default and immich renamed files. And maybe you are still using a destructive template? You could migrate to use another template.

**Comment by christerdk** [comment]
Better sorting options under Photos would be great, including "Upload time". 

I just accidentally backed up the wrong Album from my phone 🤦🏼‍♂️ (because of #16673). Some 40 photos scattered over the timeline, I don't know how to easily find them again and remove them. I guess I'll have to do it file by file based on name for now.

**Comment by goremykin** [comment]
I think we should sort by file name by default for groups of files with the same date and time. This often happens when you take old photos from a camera and set them to a date without time (00:00), because you can't remember the exact time. In this case, Immich displays the photos in random order. Although the file names in this case are perfect for sorting.

**Comment by SubjectGinger** [comment]
Big upvote on this one. Very frustrating that albums are all out of wack because there's no way to sort by filename. Especially with photos that don't have the correct EXIF data like most professional photo shoots that require editing and downloading as a package, so the files all share the same date and almost identical time.

**Comment by Cyzoc** [comment]
Just to add onto this, having the ability to sort photos in the search panel by date would be great. I find that when I use the search function, I know roughly when the photo was taken and use the context to filter everything, then scroll to the date of the photo. Personally It's really frustrating that the search feature just serves photos randomly, I can take two of the same photos, search for the same thing and something completely random and unrelated to the search comes up between the two duplicates. This wouldn't really matter to me if they were served by date, because it would be easier to skim the results. Thanks guys, immich is by far the best thing I have ever gotten!

**Comment by kokovtsev** [comment]
In addition to the above, the sort order is inconsistent between clients. I have an album where exact capture timestamps were lost so I set the same date to them in batch. Web version seems to correctly sort by timestamp AND file name, so the order of images matches the actual shooting order; whereas mobile (Android) shuffles the order completely. This happens in both timeline and album views.
